packages/zcli-themes/src/lib/pollJobStatus.ts

```diff
@@ -0,0 +1,55 @@
+import type { Job, JobError, ValidationErrors } from '../types'
+import { CliUx } from '@oclif/core'
+import { error } from '@oclif/core/lib/errors'
+import { request } from '@zendesk/zcli-core'
+import * as chalk from 'chalk'
+import validationErrorsToString from './validationErrorsToString'
+
+export default async function pollJobStatus (themePath: string, jobId: string, interval = 1000, retries = 10): Promise<void> {
+  CliUx.ux.action.start('Polling job status')
+
+  while (retries) {
+    // Delay issueing a retry
+    await new Promise(resolve => setTimeout(resolve, interval))
+
+    const response = await request.requestAPI(`/api/v2/guide/theming/jobs/${jobId}`)
```

Comment:
Would it make sense to handle if this request fails in a try / catch block?

Additionally (⚠️ Potential over-engineering alert ⚠️ ) as a small nit:

... another thing that occurs to me is the possibility that the job object might not yet be created when we call this endpoint, i.e. some form of race condition.  Unlikely but maybe a possibility?

Anyway feel free to dismiss this further suggestion but wrapping this in a try/catch along with having a separate retry variable (eg jobRetries) and trying maybe up to three times maybe with exponential backoff in a setTimeout could be a way to approach this.